geth/jail/jail_cell_test.go

```diff
@@ -109,3 +107,190 @@ func (s *JailTestSuite) TestJailLoopInCall() {
 		require.Fail("Failed to received event response")
 	}
 }
+
+func (s *JailTestSuite) TestJailLoopRace() {
```

Comment:
@adambabik right. So we start 100 Calls to `callRunner` which all will be executed in the loop and it involves potentially concurrent calls to underlying vm's functions as Set/Get/Call and Run (issue explained here: https://github.com/status-im/status-go/issues/289).
So this test, running with `-race` flag could help capture those issues.

Reply:
I see, thanks it's clear now. Just one comment regarding `-race`. Isn't it supposed to figure out race conditions by itself without having to wrap calls in a `for` loop?

Reply:
@adambabik huh, unfortunately, no. It will catch the race conditions withing the code covered by tests. That's why `go build`/`go run` also have `-race` flag - to catch races under realistic workload.

Reply:
:+1 it's described well in this blog post https://blog.golang.org/race-detector